File: Arrays/arrayOperations.js
const stringArray = ["a", "b", "c", "d"];

//push - add to the end
stringArray.push("e"); //O(1)

//pop - delete from end
stringArray.pop(); //O(1)

//unshift - add to the begining
stringArray.unshift("z"); //O(n)

console.log(stringArray);
//splice - add to specific position
//array.splice(startIndex, deleteCount, item1, item2, ...)
stringArray.splice(2, 0, "boo"); //O(n)
console.log(stringArray);

File: Arrays/ArrayOperations.java
import java.util.ArrayList;

public class ArrayOperations{
    public static void main(String[] args){
        ArrayList<String> stringArray = new ArrayList<>(); //create the array

        //add elements
        stringArray.add("a");
        stringArray.add("b");
        stringArray.add("c");
        stringArray.add("d");

        //add - add to the end
        stringArray.add("e"); //O(1)

        //remove - delete from end
        stringArray.remove(stringArray.size()-1); //O(1)

        //add - add to the begining
        stringArray.add(0,"z"); //O(n)

        System.out.println(stringArray);

        //add - add to specific position
        stringArray.add(2,"boo");

        System.out.println(stringArray);
    }
}